src/project.coffee

```diff
@@ -39,6 +39,14 @@ class Project extends Model
     @emitter = new Emitter
     @buffers ?= []
 
+    @directoryProviders = [new DefaultDirectoryProvider()]
+    atom.packages.serviceHub.consume(
+      'atom.directory-provider',
+      '^0.1.0',
+      # New providers are added to the front of @directoryProviders because
+      # DefaultDirectoryProvider is a catch-all that will always provide a Directory.
+      (provider) => @directoryProviders.splice(0, 0, provider))
```

Comment:
I agree with @maxbrunsfeld that if this put is in a spot where you had to
use a file:// uri instead of a path everywhere, that would be very
frustrating.

On Thursday, February 19, 2015, Nathan Sobo <notifications@github.com>
wrote:

> In src/project.coffee
> <https://github.com/atom/atom/pull/5535#discussion_r25039685>:
>
> > @@ -39,6 +39,14 @@ class Project extends Model
> >      @emitter = new Emitter
> >      @buffers ?= []
> >
> > +    @directoryProviders = [new DefaultDirectoryProvider()]
> > +    atom.packages.serviceHub.consume(
> > +      'atom.directory-provider',
> > +      '^0.1.0',
> > +      # New providers are added to the front of @directoryProviders because
> > +      # DefaultDirectoryProvider is a catch-all that will always provide a Directory.
> > +      (provider) => @directoryProviders.splice(0, 0, provider))
>
> @bolinfest <https://github.com/bolinfest> You're largely responsible for
> pushing me toward avoiding the requirement of running code to make these
> kinds of decisions. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/pull/5535/files#r25039685>.
>